pkg/controller/workflow/delete_handler.go

```diff
@@ -0,0 +1,30 @@
+package workflow
+
+import (
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+type DeleteEventHandler struct {
+	*handler.EnqueueRequestForObject
+	Parent interface {
+		Delete(runtime.Object) error
+	}
+}
+
+func (d *DeleteEventHandler) Delete(e event.DeleteEvent, _ workqueue.RateLimitingInterface) {
+	objectKey := kube.ObjectKey(e.Meta.GetNamespace(), e.Meta.GetName())
+	log := zap.S().With("resource", objectKey)
+
+	log.Infow("Cleaning up Atlas resource", "resource", e.Object)
```

Comment:
[nit] do you think we need both "starting" log messages here and in `Delete` method?